### PR TITLE
AAR Packaging for Android

### PR DESCRIPTION
**Pull request by mlc**
This PR:
* bundles an android library and demo app into the `swarm-icons` repo
* modifies the `dist:xml` task to build into that library instead of the old `dist/xml` directory
* and publishes the library to [our nexus repo](https://nexus.blt.meetup.com/) in travis builds

Future work:
* publish the sample app somewhere so designers can check that their icon changes are working right
* consider ways to shrink down the library size a bit?
* actually use the library in the `meetup-android` app

### CODE REVIEW

File: dist/android/app/src/main/java/com/meetup/swarmicons/IconAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.item_icon.view.*
+import java.text.Collator
+import kotlin.properties.Delegates
+
+typealias DrawableInfo = Pair<String, Int>
+
+class IconAdapter(context: Context) : RecyclerView.Adapter<IconHolder>() {
+    private val drawables : List<DrawableInfo> = R.drawable::class.java.declaredFields
+            .filter { field -> field.name.run { startsWith("ic_") && !startsWith("ic_launcher_") } }
```
Comment (mlc) [inline dist/android/app/src/main/java/com/meetup/swarmicons/IconAdapter.kt:17]: this is a hack that wouldn't be acceptable in a real app but is fine for this kind of demo

File: dist/android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,29 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    compileSdkVersion 26
+    buildToolsVersion "26.0.2"
+    defaultConfig {
+        applicationId "com.meetup.swarmicons"
+        minSdkVersion 21
```
Comment (pt2121) [inline dist/android/app/build.gradle:10]: should the demo support older versions?
or the icons will look the same anyway.

File: .travis.yml [hunk 3]
```diff
@@ -14,15 +18,28 @@ branches:
...
+
 before_script:
   - git diff --exit-code
-  - yarn run docs  
+  - yarn run docs
+
+script:
+  - export VERSION_NAME=`make version`
+  - npm test
+  - (cd dist/android && ./gradlew build icons:upload)
```
Comment (pt2121) [inline .travis.yml]: not familiar with maven plugin but i assume this is `uploadArchives`.
may be we should use the full `uploadArchives`?

File: dist/android/app/src/main/java/com/meetup/swarmicons/IconAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.item_icon.view.*
+import java.text.Collator
+import kotlin.properties.Delegates
+
+typealias DrawableInfo = Pair<String, Int>
+
+class IconAdapter(context: Context) : RecyclerView.Adapter<IconHolder>() {
+    private val drawables : List<DrawableInfo> = R.drawable::class.java.declaredFields
```
Comment (pt2121) [inline dist/android/app/src/main/java/com/meetup/swarmicons/IconAdapter.kt:16]: 👍 
just knew we can do this

**Review by pt2121 — COMMENTED** [review]
i can't build the project but android code lgtm

File: dist/android/app/build.gradle (hunk 2 above)
Comment (mlc) [reply dist/android/app/build.gradle:10]: if there's ever a reason we can change this, but then the activity has to start using appcompat and it's like an hour of work which i didn't want to do

File: .travis.yml (hunk 3 above)
Comment (mlc) [reply .travis.yml]: ok, changed

File: dist/android/app/build.gradle (hunk 2 above)
Comment (pt2121) [reply dist/android/app/build.gradle:10]: ok 👍

**Comment by mlc** [comment]
prat & mike say this is good. merging.

**Pull request merged**